### PR TITLE
Add cluster label datasource

### PR DESCRIPTION
**Pull request by coleenquadros**
This PR is to make sure the slack alert urls point to the right data source in Grafana when alert is fired

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -14,6 +14,7 @@ require (
 	github.com/alecthomas/template v0.0.0-20190718012654-fb15b899a751 // indirect
 	github.com/alecthomas/units v0.0.0-20211218093645-b94a6e3cc137 // indirect
 	github.com/beorn7/perks v1.0.1 // indirect
+	github.com/brancz/gojsontoyaml v0.1.0 // indirect
```
Comment (saswatamcode) [inline go.mod:17]: Is this dep change needed?

File: resources/observability/prometheusrules/observatorium-custom-metrics-production.prometheusrules.yaml [hunk 2]
```diff
@@ -13,7 +13,7 @@ spec:
     rules:
     - alert: ObservatoriumNoStoreBlocksLoaded
       annotations:
-        dashboard: https://grafana.app-sre.devshift.net/d/no-dashboard/observatorium-metrics?orgId=1&refresh=10s&var-datasource={{$labels.cluster}}-prometheus&var-namespace={{$labels.namespace}}&var-job=All&var-pod=All&var-interval=5m
+        dashboard: https://grafana.app-sre.devshift.net/d/no-dashboard/observatorium-metrics?orgId=1&refresh=10s&var-namespace={{$labels.namespace}}&var-job=All&var-pod=All&var-interval=5m
```
Comment (saswatamcode) [inline resources/observability/prometheusrules/observatorium-custom-metrics-production.prometheusrules.yaml:16]: Why don' we just use prometheus externalLabels here: https://prometheus.telemeter-prod-01.devshift.net/config or https://prometheus.telemeter-prod-01.devshift.net/config?
```suggestion
        dashboard: https://grafana.app-sre.devshift.net/d/no-dashboard/observatorium-metrics?orgId=1&refresh=10s&var-datasource={{$externalLabels.cluster}}-prometheus&var-job=All&var-pod=All&var-interval=5m
```

**Review by saswatamcode — COMMENTED** [review]
Thanks for taking this up! 🙂 
Some suggestions!

File: resources/services/observatorium-metrics-template.yaml [hunk 3]
```diff
@@ -265,6 +265,8 @@ objects:
       app.kubernetes.io/component: alertmanager
       app.kubernetes.io/name: alertmanager
       app.kubernetes.io/part-of: observatorium
+      cluster_env: ${CLUSTER_ENVIRONMENT}
```
[the remark this answers is not in the document]
Comment (coleenquadros) [reply resources/services/observatorium-metrics-template.yaml]: Reverted the changed from the previous commit

**Comment by saswatamcode** [comment]
Awesome! I think the SLO file changes won't be needed, so we can skip! 🙂

**Pull request merged**